packages/server/src/auth.ts

```diff
@@ -1,62 +1,109 @@
 const { GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET, GITHUB_REDIRECT_URL } = process.env;
 const GENERIC_ERROR_STRING = 'An unexpected login error has occurred.';
 
-export interface IGithubApiAccessTokenResponse {
+export interface GithubApiAccessTokenResponse {
   access_token: string;
   error: string;
   error_description: string;
 }
 
-export function getAccessTokenOrErrorResponse(
-  input: IServerAuthRequest,
-): Promise<IServerAuthResponse> {
-  let postRequest = new Promise((resolve, reject) => {
-    const url = 'https://github.com/login/oauth/access_token';
-    const data = {
-      client_id: GITHUB_CLIENT_ID,
-      client_secret: GITHUB_CLIENT_SECRET,
-      redirect_uri: GITHUB_REDIRECT_URL,
-      code: input.code,
-      state: input.state,
-    };
-
-    let request = new XMLHttpRequest();
-
-    request.addEventListener('readystatechange', () => {
-      if (request.readyState === 4 && request.status === 200) {
-        let responseData: IGithubApiAccessTokenResponse = JSON.parse(
-          request.responseText,
-        );
-        resolve(responseData);
-      } else if (request.readyState === 4) {
-        reject('Failure retrieving data');
-      }
-    });
-
-    request.open('POST', url);
-    request.setRequestHeader('Accept', 'application/json');
-    request.send(data);
-  });
-  return postRequest
-    .then(response => {
-      return getResultObjectBasedOnAuthResponse(
-        response as IGithubApiAccessTokenResponse,
-      );
-    })
-    .catch(() => {
-      return { error: GENERIC_ERROR_STRING };
-    });
-
-  // Helper
-  function getResultObjectBasedOnAuthResponse(
-    body: IGithubApiAccessTokenResponse,
-  ): IServerAuthResponse {
-    if (body.error) {
-      return { error: body.error + ': ' + body.error_description };
-    } else if (body.access_token) {
-      return { access_token: body.access_token };
-    } else {
-      return { error: GENERIC_ERROR_STRING };
+const https = require('https');
+
+async function github_auth_post(code: string, state: string) {
+  // https://github.com/login/oauth/access_token
+
+  const options = {
+    host: 'github.com',
+    method: "POST",
+    path: "/login/oauth/access_token",
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
     }
   }
+
+  const post_data = {
+    client_id: GITHUB_CLIENT_ID,
+    client_secret: GITHUB_CLIENT_SECRET,
+    redirect_uri: GITHUB_REDIRECT_URL,
+    code,
+    state,
+  };
+  const data = JSON.stringify(post_data);
+
+
+  // Good to know what is actually being sent
```

Comment:
Are you going to remove these commented out logs?  Also line 71 and line 96 has an uncommented console.log.

Reply:
I'll leave in the logs that are helpful for debugging.